packages/frontend/src/features/sqlRunner/store/barChartSlice.ts

```diff
@@ -20,16 +20,26 @@ export const barChartConfigSlice = createSlice({
                     });
 
                 state.options =
-                    sqlRunnerResultsTransformer.getCartesianLayoutOptions();
+                    sqlRunnerResultsTransformer.getPivotChartLayoutOptions();
 
-                const { newConfig, newDefaultLayout } =
-                    sqlRunnerResultsTransformer.getChartConfig({
-                        chartType: ChartKind.VERTICAL_BAR,
-                        currentConfig: state.config,
+                const { defaultLayout, mergedLayout } =
+                    sqlRunnerResultsTransformer.mergePivotChartLayout({
+                        currentConfig: state.config?.fieldConfig,
                     });
 
-                state.config = newConfig;
-                state.defaultLayout = newDefaultLayout;
+                if (state.config) {
+                    state.config.fieldConfig = mergedLayout;
+                } else {
+                    state.config = {
```

Comment:
I had moved this setting of default props out of the slice and into the 'Model' cause I thought it made more sense for the defaults to live on the models and the slices would just manage setting state. Which is why I was returning the new config from there and just setting it here. I don't have a strong reason for that except that the model seems like the right place for the defaults to be defined. At the moment the important one is just `type`, but it's definitely possible the display will also get a more complex default and having these view properties live in the slice doesn't seem quite right.